util/subnetworkid/subnetworkid_test.go

```diff
@@ -0,0 +1,443 @@
+		{"inverted", []*SubnetworkID{SubnetworkIDStartFF, SubnetworkID3, SubnetworkID2, SubnetworkID1, SubnetworkIDEndFF, SubnetworkID0}, []*SubnetworkID{SubnetworkID0, SubnetworkIDEndFF, SubnetworkID1, SubnetworkID2, SubnetworkID3, SubnetworkIDStartFF}},
+		{"shuffled", []*SubnetworkID{SubnetworkID2, SubnetworkIDEndFF, SubnetworkID3, SubnetworkID0, SubnetworkIDStartFF, SubnetworkID1}, []*SubnetworkID{SubnetworkID0, SubnetworkIDEndFF, SubnetworkID1, SubnetworkID2, SubnetworkID3, SubnetworkIDStartFF}},
+		{"with duplicates", []*SubnetworkID{SubnetworkIDEndFF, SubnetworkID2, SubnetworkID3, SubnetworkIDEndFF, SubnetworkID0, SubnetworkID1, SubnetworkID1, SubnetworkIDStartFF}, []*SubnetworkID{SubnetworkID0, SubnetworkIDEndFF, SubnetworkIDEndFF, SubnetworkID1, SubnetworkID1, SubnetworkID2, SubnetworkID3, SubnetworkIDStartFF}},
+	}
+
+	for _, test := range tests {
+		sort.Slice(test.SubnetworkIDes, func(i, j int) bool {
+			return Less(test.SubnetworkIDes[i], test.SubnetworkIDes[j])
+		})
+
+		if !reflect.DeepEqual(test.SubnetworkIDes, test.expected) {
+			t.Errorf("unexpected Sort result for"+
+				" test \"%s\". Expected: %v, got: %v.", test.name, test.expected, test.SubnetworkIDes)
+		}
+	}
+}
+
+func SubnetworkIDFlipBit(SubnetworkID SubnetworkID, bit int) SubnetworkID {
+	word := bit / 8
+	bit = bit % 8
+	SubnetworkID[word] ^= 1 << bit
+	return SubnetworkID
+}
+
+func TestSubnetworkID_Cmp(t *testing.T) {
+	r := rand.New(rand.NewSource(1))
+
+	for i := 0; i < 100; i++ {
+		SubnetworkID := SubnetworkID{}
+		n, err := r.Read(SubnetworkID[:])
+		if err != nil {
+			t.Fatalf("Failed generating a random SubnetworkID '%s'", err)
+		} else if n != len(SubnetworkID) {
+			t.Fatalf("Failed generating a random SubnetworkID, expected reading: %d. instead read: %d.", len(SubnetworkID), n)
+		}
+		SubnetworkIDBig := ToBig(&SubnetworkID)
+		// Iterate bit by bit, flip it and compare.
+		for bit := 0; bit < IDLength*8; bit++ {
+			New := SubnetworkIDFlipBit(SubnetworkID, bit)
+			if SubnetworkID.Cmp(&New) != SubnetworkIDBig.Cmp(ToBig(&New)) {
+				t.Errorf("SubnetworkID.Cmp disagrees with big.Int.Cmp New: %s, SubnetworkID: %s", New, SubnetworkID)
+			}
+		}
+		for bit := 0; bit < IDLength*8; bit++ {
+			flippedFromLeft := SubnetworkIDFlipBit(SubnetworkID, bit)
+			flippedFromRight := SubnetworkIDFlipBit(SubnetworkID, IDLength*8-bit-1)
+			if flippedFromLeft.Cmp(&flippedFromRight) != ToBig(&flippedFromLeft).Cmp(ToBig(&flippedFromRight)) {
+				t.Errorf("SubnetworkID.Cmp disagrees with big.Int.Cmp flippedFromLeft: %s, flippedFromRight: %s", flippedFromLeft, flippedFromRight)
+			}
+		}
+	}
+}
+
+func BenchmarkSubnetworkID_Cmp(b *testing.B) {
+	SubnetworkID0, err := NewFromStr("3333333333333333333333333333333333333333")
+	if err != nil {
+		b.Fatal(err)
+	}
+	for n := 0; n < b.N; n++ {
```

Comment:
It is negligible most of the time, but I still prefer to b.ResetTimer() between the setup anf the benchmark loop